api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -218,6 +222,26 @@ private User initializeUserIfNeeded() {
         throw new ConflictException("Failed due to concurrent modification");
       }
     }
+
+    // On subsequent sign-ins, complete the setup of the FC billing project. There is an
+    // initialization period for Firecloud projects, so it is not possible to do this immediately
+    // after project creation (on first sign-in).
+    if (BillingProjectStatus.PENDING.equals(user.getFreeTierBillingProjectStatus())) {
+      // If the user has not been granted the BQ job user Google role on the billing project yet,
+      // give it to them (so they can run BQ queries from notebooks).
+      try {
+        fireCloudService.grantGoogleRoleToUser(user.getFreeTierBillingProjectName(),
+            FireCloudService.BIGQUERY_JOB_USER_GOOGLE_ROLE, user.getEmail());
+      } catch (ApiException e) {
```

Comment:
The assumption here is that as soon as you can grant the role to the user, the billing project is ready for use to create workspaces. That may be true, but I don't know for a fact that this is the case -- you should check with Firecloud folks.

If that isn't the case, we may want to get the billing project status instead here.

Reply:
Good point. I didn't understand that FC was actually intentionally giving us this billing project information as kind of an endpoint to poll. I switched to checking that. I also switched to handle the case where the free tier billing project instead enters the error state.

I've also reconciled this enum with the one we use to report statuses from the billing endpoint. In the future, if we're going to have our own initialization process on workspaces, we may want to maintain some kind of state on our end as to whether or not we have initialization particular billing projects.

Lastly, I fixed a bug in firecloud.yaml; where names did not match those expected.

PTAL